src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -531,3 +531,27 @@ def _validate_and_return_run_name(run: Union[str, "Run"]) -> str:
         elif isinstance(run, str):
             return run
         raise InvalidRunError(f"Invalid run {run!r}, expected 'str' or 'Run' object but got {type(run)!r}.")
+
+    def _generate_output_path(self) -> Path:
+        config = Configuration.get_instance()
```

Comment:
@brynn-code has pointed out that user can override config with `PFClient(config={})`, in this case, the value is not available from singleton, which reads from file, so we need to handle this.